testbed/tests/widgets/properties.py

```diff
@@ -56,44 +81,69 @@ async def test_font(widget, probe):
 
     # Font family has been changed
     new_family_font = probe.font
-    assert new_family_font.family == FANTASY
+    assert_font_family(new_family_font.family, FANTASY)
 
     # Font size hasn't changed
-    assert new_family_font.size == orig_font.size * 2
-    # Button should still be taller and wider than the original
+    assert new_family_font.size == new_size_font.size
+    # Widget should still be taller and wider than the original
     assert probe.width > orig_width
     assert probe.height > orig_height
 
+    # Reset to original family and size.
+    del widget.style.font_family
+    del widget.style.font_size
+    await probe.redraw()
+    assert probe.font == orig_font
+    assert probe.height == orig_height
+    assert probe.width == orig_width
+
+
+async def test_font_attrs(widget, probe):
+    "The font weight and style of a widget can be changed."
+    assert probe.font.weight == NORMAL
+    assert probe.font.style == NORMAL
+
+    for family in [SYSTEM, SERIF]:
+        widget.style.font_family = family
+        for weight in [NORMAL, BOLD]:
+            widget.style.font_weight = weight
+            for style in [NORMAL, ITALIC]:
+                widget.style.font_style = style
+                await probe.redraw()
+                assert_font_family(probe.font.family, family)
+                assert probe.font.weight == weight
+                assert probe.font.style == style
```

Comment:
This makes me wonder whether what is needed here is a set of comprehensive "font management" tests operating on a single widget (say, Label), and a basic "can a font be set" test on each widget. If basic font setting works on a widget, we don't have any particular reason to believe that weight/variants will be problematic.

Reply:
That's true if we assume that the widgets are all using a common font implementation, and using it correctly, but that's something that could easily be broken by accident in the future. So I think it's worth being comprehensive as long as it only adds a few seconds to the tests.